vision/module/location.py

```diff
@@ -34,22 +34,46 @@ def __init__(self):
         self.lower_bound = np.array(0)  # lower bound of slopes
         self.num_buckets = np.array(0)  # number of buckets applied to slopes
 
-    ## Finding Distance to Module
+        self.needsRecalc = (
+            True  # Prevents recalculation of circles, slopes, and slope grouping
+        )
+
+    ## Determining if Module is in frame
 
-    def getDistance(self):
+    def isInFrame(self) -> bool:
         """
-        Finds the distance to the module.
+        Determines if the Module is in the frame
 
         Returns
         -------
-        int - distance to the module.
+        bool - true if module is in the frame and false if module is not in the frame
         """
-        self.distance = self.depth[self.center[1], self.center[0]]
-        return self.distance
+        MIN_SLOPES_IN_BUCKET = (
+            15  # Minimum number of slopes per bucket to identify the module
+        )
+        MAX_CIRCLES = 100  # maximum number of circles that are allowed to be detected before in_frame fails
+        MIN_CIRCLES = 4  # minimum number of circles needed to perform calculations
+
+        if self.needsRecalc:
+            # Circle Detection
+            self._circleDetection()
+
+        if (
+            np.shape(self.circles) < MIN_CIRCLES or np.shape(self.circles) > MAX_CIRCLES
```

Comment:
You have to index np.shape(self.circles) since self.circles is a two-dimensional array, otherwise you would be comparing a tuple to an int. should be:

`np.shape(self.circles)[0] < MIN_CIRCLES or np.shape(self.circles)[0] > MAX_CIRCLES`